cli/command/cli.go

```diff
@@ -137,7 +142,28 @@ func (cli *DockerCli) ServerInfo() ServerInfo {
 
 // ClientInfo returns the client details for the cli
 func (cli *DockerCli) ClientInfo() ClientInfo {
-	return cli.clientInfo
+	if cli.clientInfo != nil {
+		return *cli.clientInfo
+	}
+	var experimentalValue string
+	// Environment variable always overrides configuration
+	if experimentalValue = os.Getenv("DOCKER_CLI_EXPERIMENTAL"); experimentalValue == "" {
+		experimentalValue = cli.ConfigFile().Experimental
+	}
+	hasExperimental, _ := isEnabled(experimentalValue)
+
+	version := ""
+
+	if cli.client != nil {
+		version = cli.client.ClientVersion()
+	} else {
+		version = api.DefaultVersion
```

Comment:
Is this also a problem if this is called from the plugin? (as in; if there's no client, it will always return the default version, which may not match with  the daemon that is used